connector/connect/common/src/main/protobuf/spark/connect/ml.proto

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = 'proto3';
+
+package spark.connect;
+
+import "spark/connect/relations.proto";
+
+option java_multiple_files = true;
+option java_package = "org.apache.spark.connect.proto";
+
+
+message RemoteCall {
+  oneof call_type {
+    ConstructObject construct_object = 1;
+    DestructObject destruct_object = 2;
+    CallMethod call_method = 3;
+    CallFunction call_function = 4;
+  }
```

Comment:
Sorry for the confusion! The PR is not complete, I will complete it today and attach a runnable ML case demo and attach design sketch.

Reply:
@grundprinzip 

PR is updated.
The first stage of this PR is complete, it can support java-backend estimator's fit / transform / all model attributes / model summary.
I haven't tested pyspark Pipeline, but I guess it is already supported directly without other change.

TODO:
* Support pyspark CrossValidator and other meta-estimator
* Support pyspark evaluator
* remote object GC
* Some minor fixes (e.g. set model params / model uid correctly)

I will attach design doc later.

Reply:
As said before the current proto spec does to explain the intent of the message it's purely an interface to construct jvm classes on the other side. The goal for Spark Connect is not to have a 1:1 mapping of JVM method calls to Spark Connect RPC invocations but operate on a higher level abstraction. 

This becomes apparent when you're looking at the code that you're modifying on the Python side, you're explicitly mimicking the Py4J call interface.

Before you invest a lot of time into building a Py4J over GRPC lets have a better discussion on what the interface should look like, please!

Thanks for the hard work!

Reply:
sure! I will share design doc later for current design intention explanation and list other interface options for discussion.